crates/compiler/can/src/annotation.rs

```diff
@@ -17,20 +17,51 @@ use roc_types::types::{
 pub struct Annotation {
     pub typ: Type,
     pub introduced_variables: IntroducedVariables,
-    pub references: VecSet<Symbol>,
+    pub references: AnnotationReferences,
     pub aliases: VecMap<Symbol, Alias>,
 }
 
+#[derive(Clone, Debug)]
+pub struct AnnotationReferences {
+    symbols: VecSet<Symbol>,
+    qualified: Vec<QualifiedReference>,
```

Comment:
Is there a reason this isn't represented as `(Vec)Map<Symbol, QualifiedReference>`? It seems that would be easier to maintain for the datastructure.

Reply:
Alternatively, using `References` instead of `AnnotationReferences` here, since it appears its consumption is only into `References`.

Reply:
Good point. IIRC I thought I'd have multiple things per symbol, but it looks like that wasn't the case after all. I'll refactor it!

Reply:
I just realized that I'm tracking whether a type was referenced qualified or unqualified here, but  an annotation could have both kinds in different places. 

I already added that distinction to `References`, so I'm going try using that here as you suggested.

Reply:
Adressed [here](https://github.com/roc-lang/roc/pull/6658/commits/e500d664fdd5b79b004ec5be4614424557b67425)